examples/terraform/aws/variables.tf

```diff
@@ -16,12 +16,59 @@ variable "num_workers" {
   default     = 1
 }
 
-variable "instance_type" {
+variable "ami_id" {
+  description = "AMI ID to use for talos nodes, if not set the latest talos release ami id will be looked up"
+  type        = string
+  default     = ""
+  validation {
+    condition     = length(var.ami_id) > 0 ? (length(var.ami_id) > 4 && substr(var.ami_id, 0, 4) == "ami-") : true
+    error_message = "The image_id value must be a valid AMI id, starting with \"ami-\"."
+  }
+}
+
+variable "instance_type_control_plane" {
   description = "Instance type to use for the control plane nodes"
   type        = string
   default     = "c5.large"
 }
 
+variable "instance_type_worker" {
+  description = "Instance type to use for the worker nodes"
+  type        = string
+  default     = "c5.large"
+}
+
+variable "iam_instance_profile_control_plane" {
+  description = "IAM instance profile to use for the control plane nodes, if not set an IAM profile will not be attached"
```

Comment:
Are there known requirements for the IAM group? If so, we should at least provide a link as to what should be configured if it's going to have to be pre-created.

Reply:
probably only if deploying a CCM, I'll find some links, for now I want to integrate this to CI :sweat_smile: